### PR TITLE
[gh-#1339] fix env variables bug

### LINKED ISSUE #1339: Env variable values not updating when switching environments
> here is an example of a situation in the dashboard that could result in serious bugs:
> I have a CUSTOM_PAGE_FETCH_SERVICE_BASE_URL prod env variable set to one value
> I switch to dev environment
> The form doesn't change, and shows the prod env variable
> Only when I reload the page do I see the correct dev env variable

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Fixes a bug with how environment variables are loaded when the project environment is switched

## Issue ticket number and link
Resolves #1339 

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: n/a
- [ ] I observability, otherwise the reason is: n/a
- [ ] I added analytics, otherwise the reason is: n/a

### CODE REVIEW

File: packages/webapp/src/pages/ProjectSettings.tsx [hunk 1]
```diff
@@ -179,7 +179,12 @@ export default function ProjectSettings() {
         const entries = Array.from(formData.entries());
 
         const envVariablesArray = entries.reduce((acc, [key, value]) => {
-            const match = key.match(/^env_var_(name|value)_(\d+)$/);
+            // we use the index to match on the name and value
+            // but strip everything before hte dash to remove the dynamic aspect
```
Comment (TBonnin) [inline packages/webapp/src/pages/ProjectSettings.tsx]:
```suggestion
            // but strip everything before the dash to remove the dynamic aspect
```

File: packages/webapp/src/pages/ProjectSettings.tsx [hunk 2]
```diff
@@ -207,10 +212,11 @@ export default function ProjectSettings() {
     };
 
     const handleRemoveEnvVariable = async (index: number) => {
+        console.log(envVariables);
```
Comment (TBonnin) [inline packages/webapp/src/pages/ProjectSettings.tsx]: should this be removed?

**Review by TBonnin — APPROVED** [review]

File: packages/webapp/src/pages/ProjectSettings.tsx (hunk 2 above)
Comment (khaliqgant) [reply packages/webapp/src/pages/ProjectSettings.tsx]: Thanks, https://github.com/NangoHQ/nango/pull/1351/commits/206072616b8bc4862b2c836d1f278e97d1610ec4

File: packages/webapp/src/pages/ProjectSettings.tsx (hunk 1 above)
Comment (khaliqgant) [reply packages/webapp/src/pages/ProjectSettings.tsx]: Updated with https://github.com/NangoHQ/nango/pull/1351/commits/206072616b8bc4862b2c836d1f278e97d1610ec4

**Pull request merged**